src/app/app.component.ts

```diff
@@ -41,8 +44,15 @@ export class AppComponent implements OnInit {
         let questionNodes = this.obsFactory.getQuestionNodes(pages);
         // Extract set obs
         this.obsFactory.setValues(questionNodes, adultFormObs.obs);
+
+        // Traverse to get  orders' nodes
+        this.orderAdaptor.traverse(this.form.rootNode);
```

Comment:
Create a facade to simply consumption in the order value adapter. The value adapter's ideal interface should have 3 public methods:
generateFormPayload(form: Form); // generates the orders payload 
populateForm(form: Form, payload); // populates the form with orders

Remember the traverse methods are just implementation details. Infact create an interface, call it 'ValueAdapter'. Have it declare the two methods above.